src/pkg/bundle/deploy_test.go

```diff
@@ -429,3 +430,176 @@ func TestHelmOverrideVariablePrecedence(t *testing.T) {
+				chartName:     chartName,
+			},
+			requireNoErr: true,
+		},
+		{
+			name: "with UDS_VAR",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{},
+				},
+			},
+			args: args{
+				pkgName: pkgName,
+				variables: &[]types.BundleChartVariable{
+					{
+						Name:        varName,
+						Path:        path,
+						Type:        types.File,
+						Description: "set the var from env, so source path is current working directory (eg. /home/user/repos/uds-cli/...)",
+					},
+				},
+				componentName: componentName,
+				chartName:     chartName,
+			},
+			loadEnv:      true,
+			requireNoErr: true,
+		},
+		{
+			name: "with Config",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{
+						Config: fmt.Sprintf("%s/uds-config.yaml", relativePath),
+						Variables: map[string]map[string]interface{}{
+							pkgName: {
+								varName: "test.cert",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				pkgName: pkgName,
+				variables: &[]types.BundleChartVariable{
+					{
+						Name:        varName,
+						Path:        path,
+						Type:        types.File,
+						Description: "set the var from config, so source path is config directory",
+					},
+				},
+				componentName: componentName,
+				chartName:     chartName,
+			},
+			requireNoErr: true,
+		},
+		{
+			name: "with Bundle",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{
+
+						Source: fmt.Sprintf("%s/uds-bundle-helm-overrides-amd64-0.0.1.tar.zst", relativePath),
+					},
+				},
+			},
+			args: args{
+				pkgName: pkgName,
+				variables: &[]types.BundleChartVariable{
+					{
+						Name:        varName,
+						Path:        path,
+						Type:        types.File,
+						Description: "set the var from bundle default, so source path is bundle directory",
+						Default:     "test.cert",
+					},
+				},
+				componentName: componentName,
+				chartName:     chartName,
+			},
+			requireNoErr: true,
+		},
+		{
+			name: "file not found",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{
+						Source: fmt.Sprintf("%s/uds-bundle-helm-overrides-amd64-0.0.1.tar.zst", relativePath),
+					},
+				},
+			},
+			args: args{
+				pkgName: pkgName,
+				variables: &[]types.BundleChartVariable{
+					{
+						Name:        varName,
+						Path:        path,
+						Type:        types.File,
+						Description: "set the var from bundle default, so source path is bundle directory",
+						Default:     "not-there-test.cert",
+					},
+				},
+				componentName: componentName,
+				chartName:     chartName,
+			},
+			requireNoErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Unsetenv("UDS_CERT")
+			if tc.loadEnv {
+				os.Setenv("UDS_CERT", fmt.Sprintf("%s/test.cert", relativePath))
+			}
+
+			overrideMap := map[string]map[string]*values.Options{}
```

Comment:
let's also inspect the `overrideMap` and make sure it contains the expected values. You can do this by adding another field to the table tests called "expected" or something like that